pkg/gpu/consumer.go

```diff
@@ -110,7 +114,13 @@ func (c *cudaEventConsumer) Start() {
 				key := streamKey{pid: pid, stream: header.Stream_id}
 
 				if _, ok := c.streamHandlers[key]; !ok {
-					c.streamHandlers[key] = newStreamHandler(key.pid, c.sysCtx)
+					cgroup := unix.ByteSliceToString(header.Cgroup[:])
+					containerID, err := cgroups.ContainerFilter("", cgroup)
+					if err != nil {
+						log.Errorf("error getting container ID for cgroup %s: %s", cgroup, err)
+					}
+					metadata := model.ProcessMetadata{ContainerID: containerID}
+					c.streamHandlers[key] = newStreamHandler(key.pid, c.sysCtx, metadata)
```

Comment:
For the refactor we do after this, I think the stream handler should probably receive a "process context" object that holds the container ID, the GPU assignment, etc.